records.go

```diff
@@ -1,167 +1,72 @@
 package sarama
 
-import "fmt"
-
 const (
-	unknownRecords = iota
-	legacyRecords
-	defaultRecords
-
 	magicOffset = 16
 	magicLength = 1
 )
 
 // Records implements a union type containing either a RecordBatch or a legacy MessageSet.
 type Records struct {
-	recordsType int
```

Comment:
The reason why I put this field is to make it easier to debug possible inconsistencies between what the Records should contain and what it actually contains. If we want to refactor this, we could do it in a separate PR because I don't think it's related to the problem at hand (unless I'm missing something).

Reply:
Fair enough. I built on top of my initial changes, that's why many things were missing. I added another commit to minimize the diff with master.